lib/test_queue/runner.rb

```diff
@@ -337,6 +349,7 @@ def distribute_queue
               # If we have a slave from a different test run, don't respond, and it will consider the test run done.
               sock.write("OK\n")
               remote_workers += num
+              quorum_members += num
```

Comment:
> Hmm. That would only happen if we have run out of tests to run, and we could guard on that as well.

I guess that's true. A `@queue.empty?` check would do the trick. I'll think about this a little more but my gut says it makes sense.